django/contrib/auth/base_user.py

```diff
@@ -137,3 +136,7 @@ def get_session_auth_hash(self):
         """
         key_salt = "django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash"
         return salted_hmac(key_salt, self.password).hexdigest()
+
+    @classmethod
+    def normalize_username(cls, username):
+        return unicodedata.normalize('NFKC', force_text(username))
```

Comment:
if it happens, i suggest we modify `get_username` to pass empty string as default
```py
def get_username(self):
    return getattr(self, self.USERNAME_FIELD, '')
```